cmd/bundle/remote_state.go

```diff
@@ -0,0 +1,73 @@
+package bundle
+
+import (
+	"encoding/json"
+	"errors"
+	"os"
+	"path/filepath"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/bundle/config"
+	"github.com/databricks/cli/bundle/deploy/terraform"
+	"github.com/databricks/cli/cmd/root"
+
+	"github.com/databricks/cli/bundle/phases"
+	"github.com/spf13/cobra"
+)
+
+func newRemoteStateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "remote-state",
+		Short: "Pull and print deployed state of the bundle",
+
+		PreRunE: root.MustWorkspaceClient,
+
+		// This command is currently intended only for the Databricks VSCode extension
+		Hidden: true,
+	}
+
+	var forcePull bool
+	cmd.Flags().BoolVar(&forcePull, "force-pull", false, "Skip local cache and load the state from the remote workspace")
+
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
It's best to move all of this logic into own mutator and just call it. The reason is that it seems the code below changes bundle.Config typed structure. We have an incoming change to have a dynamic configuration alongside typed one https://github.com/databricks/cli/pull/1098

Keeping them in sync only automatically works if the config changes happen within mutator scope https://github.com/databricks/cli/pull/1098/files#diff-1af933d0c7481a7386d110dd3e52e5b919ad5e77279455343bad77f6fd88a035R24-R25